spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/AutoConfigurationSorter.java

```diff
@@ -136,8 +136,7 @@ AutoConfigurationClass get(String className) {
 		}
 
 		Set<String> getClassesRequestedAfter(String className) {
-			Set<String> classesRequestedAfter = new LinkedHashSet<>();
-			classesRequestedAfter.addAll(get(className).getAfter());
+			Set<String> classesRequestedAfter = new LinkedHashSet<>(get(className).getAfter());
```

Comment:
I find personally the previous version more readable: nesting the method call in the constructor parameter makes it harder to read.

Reply:
Create the hashset with default size ,and then put all the item in it.Would not it cause the hash table resize?